src/main/scala/SNSHandler.scala

```diff
@@ -1,34 +1,104 @@
-import java.util.Base64
-
 import com.amazonaws.SdkBaseException
-import com.amazonaws.services.sns.AmazonSNSClient
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.sns.{AmazonSNS, AmazonSNSClientBuilder}
 import com.amazonaws.services.sns.model.PublishRequest
 import com.amazonaws.services.kinesis.model.Record
+import com.gu.crier.model.event.v1.{Event, EventPayload, ItemType, _}
 import org.apache.logging.log4j.scala.Logging
 
-import scala.concurrent.Future
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ExecutionContext, Future}
+import io.circe._
+import io.circe.generic.auto._
+import io.circe.parser._
+import io.circe.syntax._
+import io.circe.generic.auto
+import com.gu.contentapi.json.CirceEncoders
+import com.gu.contentatom.thrift.Atom
+import com.gu.contentapi.client.model.v1.Content
+
+object SNSHandler extends CrossAccount with Logging {
+/*
+  def payloadId : _root_.scala.Predef.String
+  def eventType : com.gu.crier.model.event.v1.EventType
+  def itemType : com.gu.crier.model.event.v1.ItemType
+  def dateTime : scala.Long
+  def payload : scala.Option[com.gu.crier.model.event.v1.EventPayload]
+ */
+  implicit val ec:ExecutionContext = ThreadExecContext.ec
+
+  implicit val encodeAtom:Encoder[Atom] = CirceEncoders.atomEncoder
+  implicit val encodeContent:Encoder[Content] = CirceEncoders.contentEncoder
+
+  implicit val encodePayload: Encoder[Option[EventPayload]] = new Encoder[Option[EventPayload]] {
+    def getPayloadData:Option[EventPayload] => Json = {
+      case Some(payload)=>
+        payload match {
+          case EventPayload.Atom(atom)=>
+            Json.obj(("atom", atom.asInstanceOf[Atom].asJson))
+          case _=>
+            Json.Null
+        }
+      case None=>
+        Json.Null
+    }
+
+    override final def apply(a: Option[EventPayload]): Json = getPayloadData(a)
+  }
+
+  implicit val encodeEvent: Encoder[Event] = new Encoder[Event] {
```

Comment:
again, fezziwig should be able to generate this for you

Reply:
I'll try that :)